create-quasar/templates/app/quasar-v2/ts-vite/BASE/src/components/EssentialLink.vue

```diff
@@ -19,8 +19,30 @@
   </q-item>
 </template>
 
-<script lang="ts">
-<% if (typescriptConfig === 'composition') { %>import { defineComponent } from 'vue';
+<% if (typescriptConfig === 'composition-setup') { %><script setup lang="ts">
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true
+  },
+
+  caption: {
+    type: String,
+    default: ''
+  },
+
+  link: {
+    type: String,
+    default: '#'
+  },
+
+  icon: {
+    type: String,
+    default: ''
+  }
+})<% } else if (typescriptConfig === 'composition') { %><script lang="ts">
```

Comment:
Let's use TS-specific features to simplify the code.
```suggestion
export interface EssentialLinkProps {
  title: string;
  caption?: string;
  link?: string;
  icon?: string;
}
const props = withDefaults(defineProps<EssentialLinkProps>(), {
  link: '#',
});<% } else if (typescriptConfig === 'composition') { %><script lang="ts">
```